File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOUpdateStatusModel.swift
//
//  IDOUpdateStatusModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get device update status event number
@objcMembers
public class IDOUpdateStatusModel: NSObject, IDOBaseModel {
    /// Firmware version number
    public var devVesion: Int
    /// 0 for normal state
    /// 1 for upgrade state
    public var state: Int

    enum CodingKeys: String, CodingKey {
        case devVesion = "dev_vesion"
        case state = "state"
    }

    public init(devVesion: Int, state: Int) {
        self.devVesion = devVesion
        self.state = state
    }

    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOUpdateStatusModel.kt
//
//  IDOUpdateStatusModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get device update status event number

open class IDOUpdateStatusModel(devVesion: Int, state: Int) : IDOBaseModel {

    /// Firmware version number
    @SerializedName("dev_vesion")
    var devVesion: Int = devVesion

    /// 0 for normal state
    /// 1 for upgrade state
    @SerializedName("state")
    var state: Int = state


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    